src/circuits/gates.jl

```diff
@@ -409,30 +437,39 @@ gate(s::String, args...; kwargs...) = gate(GateName(s), args...; kwargs...)
 # Version that accepts a dimension for the gate,
 gate(gn::GateName, dims::Tuple; kwargs...) = gate(gn; kwargs...)
 
-
 """
     combinegates(gn::GateName, s::Tuple; kwargs...)
 
 Generate a gate (matrix) given a set of operations in the gatename string
 """
 function combinegates(gn::GateName, s::Tuple; kwargs...)
-  name = string(ITensors.name(gn))
-  name = filter(x -> !isspace(x), name)
+  name = ""
+  gate1 = nothing
+  gate2 = nothing
+
+  ignore() do 
+    name = string(ITensors.name(gn))
```

Comment:
Probably here we could define
```julia
@non_differentiable ITensors.name(::Any)
```
like I do here: https://github.com/ITensor/ITensors.jl/blob/main/src/ITensorChainRules/ITensorChainRules.jl#L247 for various operations I know won't have derivatives.

Reply:
Is this something you will define in ITensors?

Reply:
We could put it in PastaQ at first just to make this PR nicer but ultimately it (or something like it, since `ITensors.name` might go away with the now `Op` design) should go in ITensors.